java/vector/src/main/java/org/apache/arrow/vector/ipc/JsonFileReader.java

```diff
@@ -98,10 +100,14 @@ public class JsonFileReader implements AutoCloseable, DictionaryProvider {
   public JsonFileReader(File inputFile, BufferAllocator allocator) throws JsonParseException, IOException {
     super();
     this.allocator = allocator;
-    MappingJsonFactory jsonFactory = new MappingJsonFactory(new ObjectMapper()
-        //ignore case for enums
-        .configure(MapperFeature.ACCEPT_CASE_INSENSITIVE_ENUMS, true)
-    );
+    //ignore case for enums
+    ObjectMapper mapper = new ObjectMapper().configure(MapperFeature.ACCEPT_CASE_INSENSITIVE_ENUMS, true);
+    mapper.registerSubtypes(
```

Comment:
I'm not sure what this is for. In the JSON integration format, extension types are typically serialized as their respective storage type with additional metadata:
https://arrow.apache.org/docs/format/Columnar.html#extension-types

Reply:
There's actually an example here (look for "extension"):
https://arrow.apache.org/docs/format/Integration.html#json-test-data-format

Reply:
Here is an integration file as generated by the integration machinery:
https://gist.github.com/pitrou/1448c0fa62db5c8564f75c5aae7d8b17

Reply:
I don't know how Jackson works, but there's no reference to "UuidType" in the generated JSON...

Reply:
If I back this part of the change out, I get the following error:

`InvalidTypeId Could not resolve type id 'TestExtensionType$UuidType' as a subtype of `org.apache.arrow.vector.types.pojo.ArrowType`: known type ids = [binary, bool, date, decimal, duration, fixedsizebinary, fixedsizelist, floatingpoint, int, interval, largebinary, largelist, largeutf8, list, map, null, struct, time, timestamp, union, utf8] (for POJO property 'type')`

The schema it writes out is as follows:

```json
"schema" : {
    "fields" : [{
      "name" : "uuid",
      "nullable" : true,
      "type" : {
        "name" : "TestExtensionType$UuidType"
      },
      "children" : [ ],
      "metadata" : [{
        "value" : "uuid",
        "key" : "ARROW:extension:name"
      },{
        "value" : "",
        "key" : "ARROW:extension:metadata"
      }]
    }]
  }
```

Maybe this should say `type: "fixedsizebinary"` instead?

Reply:
> Maybe this should say {type: {name: "fixedsizebinary"}} instead?

Yep, it should. Again, see the generated file I posted :-)

Reply:
Ah, sorry @pitrou, I didn't see your message til I'd posted mine :facepalm: will take a look